ospfd/ospf_lsa.c

```diff
@@ -3616,6 +3616,49 @@ int ospf_lsa_more_recent(struct ospf_lsa *l1, struct ospf_lsa *l2)
 	return 0;
 }
 
+int ospf_lsa_more_recent_with_trans_delay(struct ospf_lsa *l1, struct ospf_lsa *l2, int trans_delay)
+{
+	int r;
+	int x, y;
+
+	if (l1 == NULL && l2 == NULL)
+		return 0;
+	if (l1 == NULL)
+		return -1;
+	if (l2 == NULL)
+		return 1;
+
+	/* compare LS sequence number. */
+	x = (int)ntohl(l1->data->ls_seqnum);
```

Comment:
l1seqnum and l2seqnum are better variable names.  Let's avoid things like r, x, y for non counting variables